lib/jobs/obm-control.js

```diff
@@ -64,18 +62,31 @@ function obmControlJobFactory(
             assert.ok(node, "Node should exist to run OBM command");
             assert.ok(node.obmSettings, "Node should have OBM settings to run OBM command");
 
-            var settings = _.find(node.obmSettings, function(setting) {
-                return setting ? setting.service === self.obmServiceName : false;
-            });
-
-            assert.ok(settings, "Node should have OBM settings for service: " +
-                self.obmServiceName);
+            if (self.obmServiceName) {
+                self.settings = _.find(node.obmSettings, function(setting) {
+                    return setting ? setting.service === self.obmServiceName : false;
+                });
+                // throw error if the exact obm service doesn't exist
+                assert.ok(self.settings, "Node should have OBM settings for service: " +
+                    self.obmServiceName);
+            } else {
+                //find the default obm service
+                if(node.obmSettings.length === 1) {
+                    self.settings = node.obmSettings[0];
+                    logger.info("Defaulting to obmServiceName "+ self.settings.service +
+                        " for node because only one OBM service present.");
+                }else {
+                    assert.fail(true,true, "More than one OBM service assigned to this node," +
```

Comment:
yes, I just did a quick search and this seemed like a simple enough way. Any suggestions for something different?